### PR TITLE
PAAS-1584 add budget to all HA deployments

### PR DESCRIPTION
**Pull request by grosser**
@zendesk/compute @ragurney

### CODE REVIEW

File: plugins/kubernetes/README.md [hunk 1]
```diff
@@ -148,8 +148,12 @@ or set `metadata.annotations.samson/persistent_fields`
 
 Samson can add a dynamic PodDisruptionBudget by setting `metadata.annotations.minAvailable: 30%`, it calculates the ceil of this with the configured replicas.
 (also supports absolute values like `"1"`)
+
 To remove it, set to '0', deploy, delete it from the template.
 
+Samson can auto-add a PodDisruptionBudget for every Deployment by setting for example `KUBERNETES_AUTO_MIN_AVAILABLE=80%`
+users can opt-out by setting `samson/minAvailable: disabled`.
```
Comment (adammw) [inline plugins/kubernetes/README.md]: Is this a new sentence?

File: plugins/kubernetes/test/models/kubernetes/release_doc_test.rb [hunk 2]
```diff
@@ -103,6 +103,24 @@ def create!
...
+        it "add default" do
+          create!.resource_template[2][:spec][:minAvailable].must_equal 1
+        end
+
+        it "does not add for things that are not highly available anyway" do
+          doc.replica_target = 1
+          refute create!.resource_template[2]
+        end
+
+        it "can disable with false" do
```
Comment (adammw) [inline plugins/kubernetes/test/models/kubernetes/release_doc_test.rb]: "with disabled"

File: plugins/kubernetes/app/models/kubernetes/release_doc.rb [hunk 3]
```diff
@@ -90,6 +90,11 @@ def add_pod_disruption_budget
       raw_template.each do |t|
         min_available ||= t.dig(:metadata, :annotations, :"samson/minAvailable")
       end
+      return if min_available == "disabled"
```
Comment (adammw) [inline plugins/kubernetes/app/models/kubernetes/release_doc.rb]: worth making "disabled" a constant?

**Comment by grosser** [comment]
confirmed in master by deleting a budget and then redeploying without samson/minAvailable

File: plugins/kubernetes/app/models/kubernetes/release_doc.rb (hunk 3 above)
Comment (grosser) [reply plugins/kubernetes/app/models/kubernetes/release_doc.rb]: only used once ... prefer inline

**Comment by grosser** [comment]
addressed the other 2 findings

**Pull request merged**